tests/phpunit/tests/media.php

```diff
@@ -4339,6 +4330,175 @@ public function test_wp_get_loading_optimization_attributes( $context ) {
 		}
 	}
 
+	/**
+	 * Tests that wp_get_loading_optimization_attributes() returns fetchpriority=high and increases the count for arbitrary contexts in the main loop.
+	 *
+	 * @ticket 58894
+	 *
+	 * @covers ::wp_get_loading_optimization_attributes
+	 *
+	 * @dataProvider data_wp_get_loading_optimization_attributes_arbitrary_contexts
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_optimization_attributes_with_arbitrary_contexts_in_main_loop( $context ) {
+		$attr = $this->get_width_height_for_high_priority();
+
+		// Return 'lazy' if not in the loop or the main query.
+		$this->assertSame(
+			array( 'loading' => 'lazy' ),
+			wp_get_loading_optimization_attributes( 'img', $attr, $context )
+		);
```

Comment:
```suggestion
		$this->assertSame(
			array( 'loading' => 'lazy' ),
			wp_get_loading_optimization_attributes( 'img', $attr, $context ),
			'The "loading" attribute should be "lazy" when not in the loop or the main query.'
		);
```
As this test contains multiple assertions, per [Writing PHPUnit Tests - Assertions](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions), it should have a final `$message` argument. We can remove the inline comment as the `$message` argument will provide enough context.